demisto_sdk/commands/common/tools.py

```diff
@@ -2818,9 +2823,75 @@ def compare_context_path_in_yml_and_readme(yml_dict, readme_content):
     return different_contexts
 
 
-def write_yml(yml_path: str, yml_data: Dict):
-    with open(yml_path, "w") as f:
-        yaml.dump(yml_data, f)  # ruamel preservers multilines
+def safe_write_unicode_yml_or_json(
+    path: Union[Path, str],
+    data: Dict,
+    handler: XSOAR_Handler,
+    indent: int = 0,
+    sort_keys: bool = False,
+    **kwargs,
+):
+    """
+    Write unicode content into a json/yml file.
+    """
+    path = Path(path)
+
+    if isinstance(handler, YAML_Handler):
```

Comment:
Can we keep handler default=`None`, and take the `path.suffix` to use the appropriate default handler, instead of forcing users to pass the right one? Can save us code and confusion.

Reply:
The reason I chose this way is because I am not sure that the suffix of the dumped file will always be `.json` or `.yml`

this is a better way to make sure we won't have unnecessary bugs IMO

Reply:
will use only `write_dict` for all